### ISSUE TITLE
defaultLimit = 0 don't works with a sorting param

### ISSUE THREAD

**Issue by elrancid** [balderdashy/waterline-sequel#62]
I want to disable the defaultLimit option, so for now I've set defaultLimit:-1 in bluprints config (according to this [issue](https://github.com/balderdashy/sails/issues/3224)). If I make a ajax GET request "mymodel" it works, but if I add a sort option and make GET "mymodel?sort=name ASC" the query was composed with "... ORDER BY "mymodel"."name" ASC LIMIT 0  OFFSET 0" and it don't show rows.
I've found a solution changing the line 821 in [waterline-sequel/sequel/lib/criteriaProcessor.js](https://github.com/balderdashy/waterline-sequel/blob/master/sequel/lib/criteriaProcessor.js) from
```javascript
CriteriaProcessor.prototype.limit = function(options) {
  // Some MySQL hackery here.  For details, see:
  // http://stackoverflow.com/questions/255517/mysql-offset-infinite-rows
  if(options === null || options === undefined) {
    this.queryString += ' LIMIT 184467440737095516 ';
  }
  else {
    this.queryString += ' LIMIT ' + options;
  }
};
```
to
```javascript
...
  if(options === null || options === undefined || options === 0) {
...
```
With the defaultLimit = -1 the options value is zero, so adding " || options === 0" is enough to solve the issue.

**Comment by sailsbot** [comment]
Thanks for posting, @elrancid.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
